infrastructure/src/main/java/org/corfudb/infrastructure/NodeHealthModel.java

```diff
@@ -0,0 +1,74 @@
+package org.corfudb.infrastructure;
+
+import com.google.common.reflect.TypeToken;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import org.corfudb.format.Types.NodeMetrics;
+import org.corfudb.protocols.wireprotocol.ITypedEnum;
+
+import java.util.EnumMap;
+
+/**
+ * Node Health Model:
+ * Contains a map of metrics of the state of a node.
+ * This state is utilized by the management server to
+ * make decisions on the state of the cluster and handle
+ * failures if needed.
+ * <p>
+ * Created by zlokhandwala on 1/13/17.
+ */
+public class NodeHealthModel {
```

Comment:
I do not understand this implementation that well. We should chat to get me up to speed on all the getters/setters and how they are used.